### PR TITLE
Initial cut of primitives from piet-scene

### PR DESCRIPTION
**Pull request by dfrg**
Pulls in the primitives from the piet-scene crate, replacing the geometry and path types with those from kurbo. There are a few alterations to naming and docs and the addition of some convenience "builder" methods to making constructing various types easier.

There are more types to add to this crate-- specifically things related to fonts and text. I though I'd keep the initial PR smallish, covering just enough to serve current piet-scene needs.

I have a working branch with updates to the full piet-gpu repo to support these changes that is ready to go after this lands.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+mod brush;
+mod parse;
+mod style;
+
+/// Re-export of the kurbo 2D curve library.
+pub use kurbo;
+
+pub use blend::*;
+pub use brush::*;
+pub use parse::ParseError;
```
Comment (raphlinus) [inline src/lib.rs:27]: If the only thing in there is an error, maybe the module should be named "error"? I'd expect parsing code. Minor nit though.

**Review by raphlinus — APPROVED** [review]
Generally looks good, thanks! Just some small nits and suggestions.

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "peniko"
+version = "0.1.0"
+license = "MIT/Apache-2.0"
+edition = "2021"
```
[the remark this answers is not in the document]
Comment (dfrg) [reply Cargo.toml:5]: Definitely will do.

File: Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+[dependencies]
+kurbo = { git = "https://github.com/linebender/kurbo" }
```
[the remark this answers is not in the document]
Comment (dfrg) [reply Cargo.toml:8]: Yes, that's the intention, but I didn't want to block on it.

File: src/brush/color.rs [hunk 4]
```diff
@@ -0,0 +1,720 @@
...
+    /// (currently) assumed.
+    pub fn rgba(r: f64, g: f64, b: f64, a: f64) -> Self {
+        let r = (r.max(0.0).min(1.0) * 255.0).round() as u8;
+        let g = (g.max(0.0).min(1.0) * 255.0).round() as u8;
+        let b = (b.max(0.0).min(1.0) * 255.0).round() as u8;
+        let a = (a.max(0.0).min(1.0) * 255.0).round() as u8;
+        Self { r, g, b, a }
+    }
+
+    /// Creates a color from hue, saturation, and lightness values.
```
[the remark this answers is not in the document]
Comment (dfrg) [reply src/brush/color.rs:66]: I'm not either, but one of the things I'd like to be able to offer in this library is reasonable support for the general set of relevant web style properties (in the same way that kurbo can parse SVG path data). This means being able to represent all of the color spaces (and then some?). If you'd prefer to limit the scope, I can do that.

File: src/blend.rs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    ColorBurn = 7,
+    HardLight = 8,
+    SoftLight = 9,
+    Difference = 10,
+    Exclusion = 11,
+    Hue = 12,
+    Saturation = 13,
+    Color = 14,
+    Luminosity = 15,
+    // Clip is the same as normal, but doesn't always push a blend group.
```
[the remark this answers is not in the document]
Comment (raphlinus) [reply src/blend.rs:37]: The original code had it in context of a full rendering pipeline, where the intent was crystal clear. Now it's in a separate crate, so without that context it's extremely confusing.

More seriously, yes, we should work toward documenting all this better. It was hard to motivate while everything was so fluid, but it feels like it's starting to converge a bit now.

**Comment by dfrg** [comment]
Cargo metadata and kurbo version have been updated.

I removed the hsl constructors for color (but kept hlc). Happy to spend some time in the future discussing high quality color support as I think this crate is now a suitable place for it.  I kept the (currently) unfortunately named parse module for now because I intend to fill it with some actual parsing machinery in the next PR.

Merging this so we can move forward!

**Pull request merged**